libs/network-stats/src/components/stats-manager/stats-manager.tsx

```diff
@@ -9,75 +11,75 @@ import { Table } from '../table';
-  className,
-}: StatsManagerProps) => {
-  const [data, setData] = useState<IStructuredStats | null>(null);
-  const [error, setError] = useState<Error | null>(null);
-
-  useEffect(() => {
-    async function getStats() {
-      try {
-        const [res1, res2] = await Promise.all([
-          fetch(statsEndpoint),
-          fetch(nodesEndpoint),
-        ]);
-        const [{ statistics }, { nodeData }] = await Promise.all([
-          res1.json(),
-          res2.json(),
-        ]);
-        const returned = { ...nodeData, ...statistics };
+const STATS_QUERY = gql`
+  query NetworkStats {
+    nodeData {
+      stakedTotal
+      totalNodes
+      inactiveNodes
+      validatingNodes
+      uptime
+    }
+    statistics {
+      status
+      blockHeight
+      blockDuration
+      backlogLength
+      txPerBlock
+      tradesPerSecond
+      ordersPerSecond
+      averageOrdersPerBlock
+      vegaTime
+      appVersion
+      chainVersion
+      chainId
+    }
+  }
+`;
 
-        if (!statistics || !nodeData) {
-          throw new Error('Failed to get data from endpoints');
-        }
+const compileData = (data?: NetworkStats) => {
+  const { nodeData, statistics } = data || {};
+  const returned = { ...nodeData, ...statistics };
 
-        // Loop through the stats fields config, grabbing values from the fetched
-        // data and building a set of promoted and standard table entries.
-        const structured = Object.entries(statsFields).reduce(
-          (acc, [key, value]) => {
-            const statKey = key as keyof IStats;
-            const statData = returned[statKey];
+  // Loop through the stats fields config, grabbing values from the fetched
+  // data and building a set of promoted and standard table entries.
+  return Object.entries(statsFields).reduce(
+    (acc, [key, value]) => {
+      const statKey = key as keyof IStats;
+      const statData = returned[statKey];
 
-            value.forEach((x) => {
-              const stat = {
-                ...x,
-                value: statData,
-              };
+      value.forEach((x) => {
+        const stat = {
+          ...x,
+          value: statData,
+        };
 
-              stat.promoted ? acc.promoted.push(stat) : acc.table.push(stat);
-            });
+        stat.promoted ? acc.promoted.push(stat) : acc.table.push(stat);
+      });
 
-            return acc;
-          },
-          { promoted: [], table: [] } as IStructuredStats
-        );
+      return acc;
+    },
+    { promoted: [], table: [] } as IStructuredStats
+  );
+};
 
-        setData(structured);
-        setError(null);
-      } catch (e) {
-        setData(null);
-        setError(e as Error);
-      }
-    }
+export const StatsManager = ({ className }: StatsManagerProps) => {
+  const { VEGA_ENV } = useEnvironment();
+  const { data, error, startPolling, stopPolling } =
+    useQuery<NetworkStats>(STATS_QUERY);
 
-    const interval = setInterval(getStats, 1000);
+  useEffect(() => {
```

Comment:
chore (blocking): You don't need this useEffect. Instead pass `pollInterval: true` to useQuery options

Reply:
unfortunately, if i do it that way, the polling permanently stops once i navigate off the home tab, and go back to it. if i call the start / stop in a useEffect, it retriggers it and it seems to be working fine.